lib/kubeclient/common.rb

```diff
@@ -310,7 +315,11 @@ def watch_pod_log(pod_name, namespace, container: nil)
     end
 
     def proxy_url(kind, name, port, namespace = '')
-      entity_name_plural = ClientMixin.pluralize_entity(kind.to_s)
+      entity_name_plural = if kind.to_s.end_with?('s')
```

Comment:
Might be better to have a `ClientMixin.plural?` method (or just let pluralize_entity do the check for you).  Note that not all words that end in 's' are plural, so that could be a potential problem.  Ironcally, "Kubernetes" is one of those words, but you don't pluralize that :).